### PR TITLE
#158 Hide hidden files

### PR DESCRIPTION
**Pull request by vessemer**
Filtering out the hidden files from the Vue frontend.

During working over this PR I've stumbled upon the unexpectable behaviour in `src/components/ImageSeries.vue`:
```javascript
[Vue warn]: Error in render: "TypeError: _vm.availableSeries is undefined"

found in

---> <ImageSeries> at src/components/ImageSeries.vue
       <OpenImage> at src/views/OpenImage.vue
         <App> at src/App.vue
           <Root>
```
Though the response of the `/api/images/available` was sustainable and in a proper format. It took a while from me to figure out that this behaviour was introduced by the PR #171, specifically by the exchange `$http` on `$axios` ([1](https://github.com/concept-to-clinic/concept-to-clinic/pull/171/files?diff=unified#diff-ff7a640b74e49c1a5b1dbb329689f4b7R102) and [2](https://github.com/concept-to-clinic/concept-to-clinic/pull/171/files?diff=unified#diff-ff7a640b74e49c1a5b1dbb329689f4b7R115) diffs). Thus I roll back these changes, of course, with no disruption from `UnhandledPromiseRejection`. The output of `docker-compose -f local.yml up vue` does not contain errors: 
```bash
vue_1            | [HPM] Proxy created: /  ->  http://interface:8000
vue_1            | > Starting dev server...
vue_1            |  DONE  Compiled successfully in 3260ms02:15:12
vue_1            | 
vue_1            | > Listening at http://0.0.0.0:8080
vue_1            | 
[]
```
## Description
The filtering is done on the backend side. Since there is nothing like `os.ishidden` I've decided to not over complicate and wrote the static function [`ImageAvailableApiView.is_hidden`](https://github.com/concept-to-clinic/concept-to-clinic/compare/master...vessemer:158_hide_hidden_files?expand=1#diff-500c04290e5206134d43cf7c48c3976bR80).

## Reference to official issue
This addresses the issue #158.

## Comparison:

screenshot from 2017-10-21 04-00-23
screenshot from 2017-10-21 03-47-12

## CLA
- [x] I have signed the CLA; if other committers are in the commit history, they have signed the CLA as well

### CODE REVIEW

File: interface/backend/api/views.py [hunk 1]
```diff
@@ -76,15 +76,28 @@ def filename_to_dict(name, location):
...
-        tree['children'] = [self.walk(os.path.join(location, dir), dir) for dir in folders]
+        tree = None
+        if not self.is_hidden(location):
+            folders, files = self.fss.listdir(location)
+            tree = {'name': dir_name, 'children': []}
+            tree['files'] = [self.filename_to_dict(filename, location)
+                             for filename in sorted(files)
+                             if not self.is_hidden(filename)]
+            tree['type'] = 'folder'
+            tree['children'] = [self.walk(os.path.join(location, dir), dir)
```
Comment (lamby) [inline interface/backend/api/views.py]: I think this needs a ``sorted``

File: interface/backend/api/views.py [hunk 2]
```diff
@@ -76,15 +76,28 @@ def filename_to_dict(name, location):
...
         """
-        folders, files = self.fss.listdir(location)
-        tree = {'name': dir_name, 'children': []}
-        tree['files'] = [self.filename_to_dict(filename, location) for filename in sorted(files)]
-        tree['type'] = 'folder'
-        tree['children'] = [self.walk(os.path.join(location, dir), dir) for dir in folders]
+        tree = None
+        if not self.is_hidden(location):
+            folders, files = self.fss.listdir(location)
+            tree = {'name': dir_name, 'children': []}
```
Comment (lamby) [inline interface/backend/api/views.py]: Why not just define ``tree = { .. }`` all in one go, like:

```
tree = {
    'name': dir_name
    'type': 'folder',
    'children': […],
    'files': […],
}
```

Assigning and then re-assigning makes me think (incorrectly) that we are *re-using* values which is misleading

Comment (vessemer) [reply interface/backend/api/views.py]: `tree` is now defined in one go, additionally, I've assumed that initial location shall not be hidden.

File: interface/backend/api/views.py (hunk 1 above)
Comment (vessemer) [reply interface/backend/api/views.py]: Corrected, thanks!

**Pull request merged**